demos/custom-tabs-example-app/src/main/java/org/chromium/customtabsdemos/EngagementSignalsActivity.java

```diff
@@ -0,0 +1,177 @@
+                    event = "NAVIGATION_ABORTED";
+                    break;
+                case CustomTabsCallback.NAVIGATION_FAILED:
+                    event = "NAVIGATION_FAILED";
+                    break;
+                case CustomTabsCallback.NAVIGATION_FINISHED:
+                    event = "NAVIGATION_FINISHED";
+                    break;
+                case CustomTabsCallback.NAVIGATION_STARTED:
+                    event = "NAVIGATION_STARTED";
+                    break;
+                case CustomTabsCallback.TAB_SHOWN:
+                    event = "TAB_SHOWN";
+                    break;
+                case CustomTabsCallback.TAB_HIDDEN:
+                    event = "TAB_HIDDEN";
+                    break;
+                default:
+                    event = String.valueOf(navigationEvent);
+            }
+            Log.d(TAG, "onNavigationEvent (navigationEvent=" + event + ')');
+            mTextNavigation.setText("onNavigationEvent " + event);
+        }
+    };
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_engagement_signals);
+        findViewById(R.id.start_custom_tab).setOnClickListener(this);
+
+        mUrlEditText = findViewById(R.id.url);
+        mTextGreatestPercentage = findViewById(R.id.label_event_greatest_percentage);
+        mTextNavigation = findViewById(R.id.label_event_navigation);
+        mTextSessionEnd = findViewById(R.id.label_event_session_ended);
+        mTextVerticalScroll = findViewById(R.id.label_event_vertical_scroll);
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        bindCustomTabsService();
+    }
+
+    private void bindCustomTabsService() {
+        String packageName = CustomTabsHelper.getPackageNameToUse(this);
+        if (packageName == null) return;
+        mConnection = new ServiceConnection(mServiceConnectionCallback);
+        CustomTabsClient.bindCustomTabsService(this, packageName, mConnection);
+    }
+
+    @Override
+    protected void onStop() {
+        super.onStop();
+        unbindCustomTabsService();
+    }
+
+    private void unbindCustomTabsService() {
+        if (mConnection == null) return;
+        unbindService(mConnection);
+        mClient = null;
+        mCustomTabsSession = null;
+        mConnection = null;
+    }
+
+    @Override
+    public void onClick(View v) {
+        int viewId = v.getId();
+        switch (viewId) {
+            case R.id.start_custom_tab:
+                openCustomTab();
+                break;
+            default:
+                // Unknown View Clicked
+        }
+    }
+
+    private void openCustomTab() {
+        String url = mUrlEditText.getText().toString();
+        CustomTabsIntent.Builder intentBuilder = new CustomTabsIntent.Builder(mCustomTabsSession);
+        intentBuilder.setInitialActivityHeightPx(INITIAL_HEIGHT_DEFAULT_PX);
```

Comment:
nit: Any reason not to use `CustomTabsIntent.ACTIVITY_HEIGHT_DEFAULT`, instead of the static import (other files seem to avoid the static import, so thought worth asking)